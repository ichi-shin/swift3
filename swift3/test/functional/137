#!/bin/bash
# Copyright(c)2014 NTT corp. All Rights Reserved.
# DB1-1-8-2_5

. ./common

S3USER=tester

_s3_put /bucket

upload_id=$(_s3_post /bucket/object?uploads \
    -H 'Content-Disposition: attachment; filename="a.txt.gz"' \
    -H "Content-Encoding: gzip" \
    -H "x-amz-meta-hoge: hoge" \
    -H "x-amz-storage-class: STANDARD" | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')

_random | dd of=$tmp.data bs=4k seek=1024 count=1 > /dev/null 2>&1
_s3_put /bucket/object?uploadId=hoge\&partNumber=1 -T $tmp.data | _filter_curl xml

