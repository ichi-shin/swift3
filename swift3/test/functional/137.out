QA output created by 137
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X POST -H Content-Disposition: attachment; filename="a.txt.gz" -H Content-Encoding: gzip -H x-amz-meta-hoge: hoge -H x-amz-storage-class: STANDARD http://SWIFT_HOST/bucket/object?uploads... 200
> s3curl --id tester -- -X PUT -T /TMP.data http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=1... 404
<?xml version='1.0' encoding='UTF-8'?>
<Error>
  <Code>NoSuchUpload</Code>
  <Message>The specified multipart upload does not exist. The upload ID might be invalid, or the multipart upload might have been aborted or completed.</Message>
  <RequestId>TXID</RequestId>
  <UploadId>UPLOAD_ID</UploadId>
</Error>
